### PR TITLE
Added Java code for Z-Algorithm

### PR DESCRIPTION
**Pull request by prashant-mahanta**
Issue: [#741](https://github.com/jainaman224/Algo_Ds_Notes/issues/741)

### CODE REVIEW

File: Z_Algorithm/Z_Algorithm.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+public class Z_Algorithm{
```
Comment (jainaman224) [inline Z_Algorithm/Z_Algorithm.java]: Add space after class name.

File: Z_Algorithm/Z_Algorithm.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	                while(Right < n && str.charAt(Right - Left) == str.charAt(Right))
+	                    Right++;
+
+	                Z[i] = Right - Left;
+	                Right--;
+	            }
+	        }
+	    }
+	}
+	static void search(String text, String pattern){
```
Comment (jainaman224) [inline Z_Algorithm/Z_Algorithm.java]: Add new line above this function.

File: Z_Algorithm/Z_Algorithm.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+
+	}
+	public static void main(String args[]){
+
+		String text = "namanchamanbomanamansanam";
+		String pattern = "aman";
+		search(text, pattern);
+	}
+}
```
Comment (jainaman224) [inline Z_Algorithm/Z_Algorithm.java]: Add sample input output.

**Comment by prashant-mahanta** [comment]
@jainaman224 I have updated the code. Please have a look at it.

**Pull request merged**